virtual_rainforest/models/abiotic_simple/abiotic_simple_model.py

```diff
@@ -0,0 +1,232 @@
+            )
+            LOGGER.error(to_raise)
+            raise to_raise
+
+        super().__init__(data, update_interval, **kwargs)
+
+        # create a list of layer roles
+        layer_roles = set_layer_roles(canopy_layers, soil_layers)
+
+        self.data
+        """A Data instance providing access to the shared simulation data."""
+        self.layer_roles = layer_roles
+        "A list of vertical layer roles."
+        self.update_interval
+        """The time interval between model updates."""
+        self.initial_soil_moisture = initial_soil_moisture
+        """Initial soil moisture for all layers and grill cells identical."""
+        self.time_index = 0
+        """Start counter for extracting correct input data."""
+
+    @classmethod
+    def from_config(
+        cls, data: Data, config: dict[str, Any], update_interval: Quantity
+    ) -> AbioticSimpleModel:
+        """Factory function to initialise the abiotic simple model from configuration.
+
+        This function unpacks the relevant information from the configuration file, and
+        then uses it to initialise the model. If any information from the config is
+        invalid rather than returning an initialised model instance an error is raised.
+
+        Args:
+            data: A :class:`~virtual_rainforest.core.data.Data` instance.
+            config: The complete (and validated) Virtual Rainforest configuration.
+            update_interval: Frequency with which all models are updated.
+        """
+
+        # Find number of soil and canopy layers
+        soil_layers = config["abiotic_simple"]["soil_layers"]
+        canopy_layers = config["abiotic_simple"]["canopy_layers"]
+        initial_soil_moisture = config["abiotic_simple"]["initial_soil_moisture"]
+
+        LOGGER.info(
+            "Information required to initialise the abiotic simple model successfully "
+            "extracted."
+        )
+        return cls(
+            data, update_interval, soil_layers, canopy_layers, initial_soil_moisture
+        )
+
+    def setup(self) -> None:
+        """Function to set up the abiotic simple model."""
+
+        # setup_variables = simple_regression.setup_simple_regression(
+        #     layer_roles=self.layer_roles,
+        #     data=self.data,
+        #     initial_soil_moisture=self.initial_soil_moisture,
+        # )
+        # update_data_object(data=self.data, output_dict=setup_variables)
+
+    def spinup(self) -> None:
+        """Placeholder function to spin up the abiotic simple model."""
+
+    def update(self) -> None:
+        """Placeholder function to update the abiotic simple model."""
+
+        # output_variables = simple_regression.run_simple_regression(
+        #     data=self.data,
+        #     layer_roles=self.layer_roles,
+        #     time_index=self.time_index,
+        # )
+        # update_data_object(data=self.data, output_dict=output_variables)
+        self.time_index += 1
```

Comment:
Ohhhh. I get this now. At the moment, the `BaseModel.update` method just says update - it doesn't say this is the update for 20th June 2014. So _any_ module that needed to refer to data with a temporal dimension would need to keep an internal tracker. At the moment, that is just `abiotic` but that is likely to change at some point.

So... @jacobcook1995 @dalonsoa I think we need to amend the `update` method to pass in the timepoint from the `vr_run` loop? That could be a datetime object that can then be used to find the matching time slice, or it could be that `vr_run` tracks the indices of the timesteps alonf the temporal axis and simply passes in that index.

Reply:
Yeah this is something we definitely need to give thought to down the line. (I would personally favour something that regularises the data such that a simple index can be used throughout).

I don't think it can be resolved in the near future as it requires us to all agree on the exact details of the timing system we will use, e.g. are we using 1 month time stamps (i.e. Jan = 31 days, Feb = 28/29 days, March = 31 days, etc), or an average length month (1 month = 30.437), or simplifying to a standard 30 day month.

I don't think we can reach a consensus on what is the best approach is until we know the needs of each model. So, I think this is definitely something to be resolved down the line.

Reply:
The temporal axis of the data drives it, I think? If we are working off a common time step, then that should be synchronised with whatever the increment is in the data. Or perhaps more pragmatically, we can use `isel` within the models to find the nearest match to the simple timestamp from the `vr_run` loop. That potentially leads to using data from the future, so we'd need to be careful. Alternatively `vr_run` just increments over the temporal axis from the data rather than using a separate update interval calculation?

Reply:
Thanks for your comments on this. I agree that in the long-term we should have a time axis and counter outside the abiotic model but for the first tests it should work. I  added a `time_index` coordinate to my dummy dataset to try this as soon as the model is complete in develop.